code-analysis/crates/solidity/outputs/typescript/lib/index.d.ts

```diff
@@ -4,10 +4,9 @@
 /* auto-generated by NAPI-RS */
 
 export const enum CSTNodeType {
-  None = 0,
-  Rule = 1,
-  Token = 2,
-  Group = 3
+  Rule = 0,
```

Comment:
Unrelated to this PR:

Can we move `index.d.ts` and `index.js` to a `typescript/lib/generated` folder? to make sure it is marked/treated as the rest of generated files. The `package.json` can just refer to it using the `main` and `types` properties:

https://docs.npmjs.com/cli/v9/configuring-npm/package-json#main

https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package

Reply:
I'm not sure I understand how these are generated, or it might have not been automated yet, but in the future, our `CodegenContext` API should take care of validating it in CI, and also formatting it automatically using `prettier` as well, so we don't need to special case them in `.prettierignore`.

Reply:
Yes, to both of these